cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -95,6 +96,19 @@ func (c *Commander) Restart(ctx context.Context) error {
 	return c.Start(ctx)
 }
 
+func (c *Commander) ReloadConfigFile() error {
+	if c.cmd == nil || c.cmd.Process == nil {
+		return errors.New("agent process is not running")
+	}
+
+	c.logger.Debug("Sending SIGHUP to agent process to reload config", zap.Int("pid", c.cmd.Process.Pid))
+	if err := c.cmd.Process.Signal(syscall.SIGHUP); err != nil {
```

Comment:
Do you know what the behavior is if the Collector doesn't handle (or doesn't reload the config for) SIGHUP signals? It's unlikely but possible we may control a Collector that has customizations that don't match upstream's behavior.

Reply:
Correct, the reloading process becomes a no-op. The signal will be sent successfully to the agent and on the next health report it will be considered healthy and ready, even though it might still be running on an old configuration from the Supervisor's pov.

Reply:
Great, thank you for the details.

Could you leave a comment describing the expected behavior in these three scenarios on the `(*Supervisor).hupReloadAgent` method?